sourcegraph/sourcegraph
# remove extension registry UI and related GraphQL API

Removes the extension registry UI and the GraphQL API that is necessary for the registry extension UI.

Any extensions that already exist locally or remotely can still be used with the enableLegacyExtensions site config flag and by enabling the extensions in your user settings.

This change is an intermediate step toward removing the extension registry (except the endpoint that's needed to support pre-4.0 Sourcegraph instances).

I think this PR has a 99% chance of going into the 4.4 release (2023-01-20) and not needing to be reverted. Subsequent PRs that actually make it impossible to use local or custom legacy extensions are still pending customer communication and confirmation. I hope we can merge other future PRs there as well, but there is a greater chance we'd need to revert  them before 4.4. By making this PR first, it means less of a revert and more progress regardless of what we learn from customers.

See [more internal context](https://sourcegraph.slack.com/archives/C03P1V3J92P/p1671668923668759) and the [tracking gdoc](https://docs.google.com/document/d/10vtoe-kpNvVZ8Etrx34bSCoTaCCHxX8o3ncCmuErPZo/edit#).

## Test plan

1. Add `"extensions": {"sourcegraph/github-extras": true}` to your user settings.
2. Visit https://sourcegraph.test:3443/github.com/sourcegraph/create/-/blob/prettier.config.js#tab=github.pullRequests.
3. Expected: a panel tab with PRs.
